trex-scripts/tests/priority_scheduling.py

```diff
@@ -0,0 +1,152 @@
+# SPDX-FileCopyrightText: Copyright 2020-present Open Networking Foundation.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import os
+import json
+from argparse import ArgumentParser
+from datetime import datetime
+from scapy.layers.all import IP, TCP, UDP, Ether
+
+from lib.base_test import StatelessTest
+from lib.utils import list_port_status,get_readable_port_stats
+from lib.xnt import analysis_report_pcap
+from trex_stl_lib.api import STLPktBuilder, STLStream, STLTXCont, STLTXSingleBurst, STLFlowLatencyStats
+
+
+SOURCE_MAC = "00:00:00:00:00:01"
+DEST_MAC   = "00:00:00:00:00:02"
+
+SENDER_PORT   = [0]
+RECEIVER_PORT = [1]
+
+# In this test case, we consider two types of flows: 1) delay critical, and 2) best effort
+# assume max. latency for delay critical flows
+LATENCY_DC_MAX_uSEC = 1000   # in mircoseconds
+# assume max. latency for best effort (i.e., low priority flows) traffic during congestion
+# with the priortization of delay critical flows
+LATENCY_LP_MAX_uSEC = 24000 # in mircoseconds
+
+
+
+class PriorityScheduling(StatelessTest):
+
+    # setup_subparser is an optional class method
+    # You can implement this method if you want to add additional command line
+    # parameters for your test.
+    # Those parameters will be parsed and be passed to the "start" method below as
+    # "args" argument.
+    @classmethod
+    def setup_subparser(cls, parser: ArgumentParser) -> None:
+        parser.add_argument("--duration", type=float, help="Test duration", default=10)
+
+
+    # The entrypoint of a test
+    def start(self, args: dict) -> None:
+
+        # create packets
+        pkt1 = Ether(dst=DEST_MAC) / IP(src="16.0.0.1",dst="48.0.0.1") / UDP(dport=12,sport=1025) / ("*" * 1500)
+        pkt2 = Ether(dst=DEST_MAC) / IP(src="16.0.0.2",dst="48.0.0.2") / UDP(dport=12,sport=1025) / ("*" * 1500)
+        pkt3 = Ether(dst=DEST_MAC) / IP(src="16.0.0.3",dst="48.0.0.3") / UDP(dport=12,sport=1025) / ("*" * 1500)
+
+        #stream list
+        streams = []
+
+        # Create a traffic stream
+        # assume s1 is a delay critical stream with QoS
+        s1 = STLStream( packet=STLPktBuilder(pkt=pkt1), mode=STLTXCont(percentage = 1), flow_stats = STLFlowLatencyStats(pg_id = 5))
```

Comment:
It would make the code more readable if you could set `pg_id` in a consistent way following the stream variable names, i.e. 1, 2, 3. So `s1` should have `pg_id = 1`, `s2` should have `pg_id = 2`, etc.